Release 1.1.0

itrujnara opened the pull request:
<!--
# nf-core/reportho pull request

Many thanks for contributing to nf-core/reportho!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the dev branch, unless you're preparing a pipeline release.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/reportho/tree/master/.github/CONTRIBUTING.md)
-->
This is the pull request for release 1.1.0. See the changelog for changes.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/reportho/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, also make a PR on the nf-core/reportho _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

atrull314 commented on CITATIONS.md:
Does this need to be updated since T-COFEE is being removed and DIAMOND is being added?

atrull314 commented on modules/local/fetch_ensembl_sequences.nf:
```diff
@@ -0,0 +1,50 @@
+    def add_query = query_fasta == [] ? "" : "cat $query_fasta >> ${prefix}_orthologs.fa"
+    """
+    fetch_ensembl_sequences.py $ids $ensembl_idmap $prefix > ${prefix}_ensembl_sequences.fa
+    $add_query
```
Does the `*_orthologs.fa` need to be added to the output directive above? You can set it as an optional output file since the `query_fasta` above is not a required input

atrull314 commented on modules/local/fetch_ensembl_sequences.nf:
```diff
@@ -0,0 +1,50 @@
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = task.ext.prefix ?: meta.id
+    def add_query = query_fasta == [] ? "" : "cat $query_fasta >> ${prefix}_orthologs.fa"
```
Will want to make sure these follow Harshil Alignment

atrull314 commented on modules/local/fetch_oma_sequences.nf:
```diff
@@ -23,7 +23,7 @@ process FETCH_SEQUENCES_ONLINE {
     def prefix = task.ext.prefix ?: meta.id
     def add_query = query_fasta == [] ? "" : "cat $query_fasta >> ${prefix}_orthologs.fa"
```
Harshil Alignment

atrull314 commented on modules/local/fetch_refseq_sequences.nf:
```diff
@@ -1,29 +1,30 @@
 
     script:
-    prefix = task.ext.prefix ?: meta.id
+    def prefix = task.ext.prefix ?: meta.id
+    def add_query = query_fasta == [] ? "" : "cat $query_fasta >> ${prefix}_orthologs.fa"
```
Harshil Alignment

atrull314 commented on modules/local/fetch_uniprot_sequences.nf:
```diff
@@ -0,0 +1,49 @@
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = task.ext.prefix ?: meta.id
+    def add_query = query_fasta == [] ? "" : "cat $query_fasta >> ${prefix}_orthologs.fa"
```
Harshil Alignment

atrull314 commented on modules/local/filter_hits.nf:
```diff
@@ -21,7 +21,7 @@ process FILTER_HITS {
 
     script:
-    prefix     = task.ext.prefix ?: meta.id
+    def prefix     = task.ext.prefix ?: meta.id
     targetfile = use_centroid ? "${prefix}_centroid.txt" : "${prefix}_minscore_${min_score}.txt"
```
Harshil Alignment

atrull314 commented on bin/fetch_ensembl_idmap.py:
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env python3
+
+# Written by Igor Trujnara, released under the MIT license
+# See https://opensource.org/license/mit for details
```
Maybe add a quick docstring describing what this script does?

atrull314 commented on bin/fetch_ensembl_sequences.py:
Maybe add a quick docstring describing what this script does? As well as docstrings for each of the functions

atrull314 commented on bin/fetch_refseq_sequences.py:
Maybe add a quick docstring describing what this script does as well as docstrings for each of the functions?

atrull314 commented on bin/fetch_uniprot_sequences.py:
Maybe add a quick docstring describing what this script does as well as docstrings for each of the functions?

atrull314 commented on bin/make_merge_table.py:
Maybe add a quick docstring describing what this script does as well as docstrings for each of the functions?

atrull314 commented on bin/split_id_format.py:
Maybe add a quick docstring describing what this script does as well as docstrings for each of the functions?

atrull314 commented on bin/fetch_uniprot_sequences.py:
```diff
@@ -0,0 +1,66 @@
+from utils import list_to_file, safe_get, SequenceInfo, split_ids
+
+
+def fetch_slice(ids: list[str]) -> list[SeqIO.SeqRecord]:
+    payload: dict[str,str] = {"accession": ','.join(ids)}
+    headers: dict[str,str] = {"Accept": "text/x-fasta"}
+    res = safe_get("https://www.ebi.ac.uk/proteins/api/proteins",
+                       params = payload,
+                       headers = headers)
+    if not res.ok:
+        return []
+
+    tmp = io.StringIO(res.content.decode())
+    seqs = SeqIO.parse(tmp, "fasta")
+
+    return list(seqs)
```
I've seen this fetch_slice function come up a couple times in files, seems like there's some shared code but still some differences. Might be worth thinking about abstracting some of the shared templates out to a 'utils' or 'helper' script for easier maintenance?

itrujnara replied:
I've already moved large chunks of code into the utils file. The functions in the database scripts should only handle the preparation of API-specific payload and then unpacking the returned information. Let me know if you spot something more that can be moved.

atrull314 replied:
Ok sure, it looked like there were some similarities in the fetch_slice that may be worth abstracting out. But I think it's also fine as is. I'll mark this as resolved

atrull314 commented:
@itrujnara: Very good PR overall! Just added a few thoughts/suggestions.

I'll also place this here. I saw there are a some of the custom python scripts use positional inputs arguments, may be worth converting those to flag-based using a package like argparse as it would help with maintenance and code readability (it would also automate some of the argument validation/checks). Not something I think is needed for this PR, but just something to think about in the future

Thanks!

itrujnara commented:
@atrull314 I have addressed the comments in PR #91. Can you please review it? I've created an issue for the Python named arguments, but I have deferred it to version 1.2.0

MeriamOs reviewed (changes requested):
This is my first time reviewing, and still relatively new to nf-core, so not very good at picking up on code bugs etc., but I picked up on a few minor things. Most importantly I think is some double up code in the nextflow.config with contrasting versions of nf-schema.

atrull314 commented:
> @atrull314 I have addressed the comments in PR #91. Can you please review it? I've created an issue for the Python named arguments, but I have deferred it to version 1.2.0 (issue #90)

@itrujnara Yep I've approved it and resolved all the issues in this PR

itrujnara commented on bin/ensembl2uniprot.py:
Looks okay on my side

itrujnara commented on CHANGELOG.md:
Nice catch

itrujnara commented on modules/nf-core/cat/cat/environment.yml:
```diff
@@ -1,7 +1,5 @@
-name: iqtree
```
This file is from nf-core/modules, I cannot edit it here

itrujnara commented on subworkflows/local/get_orthologs.nf:
I've added them to separate different kinds of modules, I think it makes long import blocks easier to understand

itrujnara commented:
@MeriamOs I have addressed your comments. The changes are in #92. Can you please review it? 
I don't understand the comments about empty files, they all look correct on my side. Can you please elaborate?

MeriamOs commented:
> @MeriamOs I have addressed your comments. The changes are in #92. Can you please review it? I don't understand the comments about empty files, they all look correct on my side. Can you please elaborate?

Weird, they appeared empty at the time I reviewed them
![image](https://github.com/user-attachments/assets/2dcd39f6-0d8d-4b00-887e-92836709ad66)

But I can see the content now, so should be fine!
